lighthouse/classes/event_properties/interfaces/event_property_abstract.py

```diff
@@ -0,0 +1,161 @@
+        """
+        self.reset()
+        self._params = params
+        if self._params is None:
+            raise ValidationError("You need to define params to create the EventProperty")
+
+    def reset(self):
+        """
+        Deletes all errors and clears the validation setting.
+        """
+        self._errors: List[str] = []
+        self._value = None
+        self._is_valid = True
+
+    @property
+    def errors(self) -> List[str]:
+        """
+        is_valids the instance and returns the complete list of errors found (this
+        errors are not just validations; it could also be previous exceptions
+        thrown during the lifetime of this instance).
+
+        Arguments:
+            None
+
+        Returns:
+            List[str] - List of error messages found currently
+        """
+        self.is_valid()
+        return self._errors
+
+    def valid(self) -> bool:
+        """Alias for #is_valid()"""
+
+        return self.is_valid()
+
+    def enforce_validation(self) -> None:
+        """
+        Raises a ValidationError exception if the instance does not pass validation.
+
+        Returns:
+            ValidationError - Raises exception
+        """
+        if not self.is_valid():
+            raise ValidationError("Validation error")
+
+    def process_validation(self, condition: bool, message: str) -> None:
+        """
+        Stores the error message if the condition is not True.
+        Changes the validation state for the instance.
+        It is recommended to run this inside a validation_scope() context
+        to be able to properly log any possible exceptions while checking the
+        condition.
+
+        Arguments:
+            condition: bool - Condition to check
+            message: str - Error message to store
+
+        Returns:
+            None
+
+        """
+        if not condition:
+            if message not in self._errors:
+                self._errors.append(message)
+            self._is_valid = False
+        self._is_valid = self._is_valid and True
+
+    @contextmanager
+    def validation_scope(self):
+        """
+        Creates a 'safe' scope where we can perform a validation check without
+        raising the exception, but log this situations if it happens.
+        This is intended to be use only with small validation operations inside the
+        is_valid() method. It is not recommended in any other case.
+
+        Arguments:
+            None
+
+        Returns:
+            ContextManager - A context specifically created to handle a validation error
+        """
+        try:
+            yield
+        except Exception as exc:
+            logger.debug(f"At { self._source_code_position_for_logging() } - Exception during validation")
+
+            self._is_valid = False
+            msg = f"Unexpected exception while trying to is_valid {exc}"
+            if msg not in self._errors:
+                self._errors.append(msg)
+            logger.exception(exc)
```

Comment:
The `.exception()` method gives the full trace back so do you need the debug line on 116?

Reply:
I'll remove it.